packages/lib-core/src/api/common-types.ts

```diff
@@ -0,0 +1,40 @@
+import type { AnyObject } from '@monorepo/common';
+
+// Type for extension hook
+export type ExtensionHook<TType, TOptions extends AnyObject = AnyObject> = (
+  options: TOptions,
+) => ExtensionHookResult<TType>;
+
+// Type for extension hook result that returns [data, resolved, error]
+export type ExtensionHookResult<TType> = [TType, boolean, unknown];
```

Comment:
`ReturnType<ExtensionHook<...>>` will return this value. I think we can avoid exporting this.

Reply:
It should see the [TypeScript Playground example](https://www.typescriptlang.org/play?#code/C4TwDgpgBAggdiA8gIwFYQMbCgXigbwF8BuKAejKgDMB7AJymAgGdgBLOAcwBpGALNsygYAhnCjJoYkMAFcAsACglFKABVw0WgwgAPJnGZsa4vjRoBrJXrD1soSFACi+iIeNwAEuYsAeNRqQvGqIYOwmQnoGACZC8EhomNh48SjoWAB8uFAAFEpQUDRhHswAXOqh4YbcSgCUuFkuBkYm3pYASiwArgA2wP6BEBnESiqUg9T0UFFuLaY+UHTdffwi2EvAXXSGUADa0WsivEvMND0AbhDRvBB0dPQAuta6tnT2ms6u7q0+ncy9-QCmiyeF2QKCEnMPQgYl4XTgFjgNAA7nAHiNlJiHNAAMo0AC2EAAwl1WASJnh8FA4CJCeVWHQOJwoCRRopVABZGik6A0S4MAAGahYwAFjBoUGYEGgshlHzY2D4ImYSmx6hF2U6m22g18TVmHjafjxhJJZPxgwyGSAA) I coded up to test it.

Reply:
We already have `UseResolvedExtensionsResult<TExtension extends Extension>` type declared in `packages/lib-core/src/store/useResolvedExtensions.ts` so we should reuse that one.

The copied-over `ExtensionHook` type actually refers to the result of `useResolvedExtensions` hook.

Reply:
I'm fine with it either way.

My thoughts are - I think we could just use the `ReturnType` for types that have no name-benefit. It's a 3-param array and is just a collection of responses. A name will not help us use it anymore than if it was a `ReturnType<...>` result.